### ISSUE TITLE
[CI flake] cy.origin window test flake

### ISSUE THREAD

**Issue by mjhenkes** [cypress-io/cypress#21376]
### Current behavior

Six tests timed out in the [cy.origin window tests](https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/integration/e2e/origin/commands/window.spec.ts)

```bash
CypressError: Timed out after waiting `60000ms` for your remote page to load on origin(s):

- `http://foobar.com:3500`

A cross-origin request for `http://www.foobar.com:3500/fixtures/dom.html` was detected.

A command that triggers cross-origin navigation must be immediately followed by a `cy.origin()` command:

`cy.origin('http://foobar.com:3500', () => {`
`  <commands targeting http://www.foobar.com:3500 go here>`
`})`

If the cross-origin request was an intermediary state, you can try increasing the `pageLoadTimeout` value in `cypress.json` to wait longer.

Browsers will not fire the `load` event until all stylesheets and scripts are done downloading.

When this `load` event occurs, Cypress will continue running commands.
    at timedOutWaitingForPageLoad (cypress:///../driver/src/cy/commands/navigation.ts:84:72)
    at eval (cypress:///../driver/src/cy/commands/navigation.ts:493:16)
    at tryCatcher (cypress:////root/cypress/node_modules/bluebird/js/release/util.js:16:23)
    at eval (cypress:////root/cypress/node_modules/bluebird/js/release/catch_filter.js:17:41)
    at tryCatcher (cypress:////root/cypress/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (cypress:////root/cypress/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (cypress:////root/cypress/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (cypress:////root/cypress/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (cypress:////root/cypress/node_modules/bluebird/js/release/promise.js:690:18)
    at _drainQueueStep (cypress:////root/cypress/node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (cypress:////root/cypress/node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (cypress:////root/cypress/node_modules/bluebird/js/release/async.js:147:5)
    at Async.drainQueues (cypress:////root/cypress/node_modules/bluebird/js/release/async.js:17:14)
From Your Spec Code:
    at Context.eval (http://localhost:3500/__cypress/tests?p=cypress/integration/e2e/multi-domain/commands/multi_domain_window.spec.ts:107:8)
```

The tests that flaked were:

* `cy.origin window .window()`
* `cy.origin window .document()`
* `cy.origin window .title()`
* `cy.origin window #consoleProps .window()`
* `cy.origin window  #consoleProps .document()`
* `cy.origin window  #consoleProps .title()`

### Desired behavior

Tests run reliably.

### Test code to reproduce

See: https://app.circleci.com/pipelines/github/cypress-io/cypress/36903/workflows/c8342f8f-3389-4af5-8a2a-0e09f5ff37ba/jobs/1481372

### Cypress Version

9.6.0

**Comment by mrkjdy** [comment]
I'm having trouble using the `cy.origin()` command and am facing the same error when trying to login to my companies SSO
```TypeScript
cy.origin('login.ibm.com', () => {
  cy.visit('/') // <- hangs here saying the page never fired its 'load' event
    .get('#username') // never gets called
    // ...
});
```
The same visit does not hang when used outside of an origin block. Should I open a different issue for this?

**Comment by idanElitzur** [comment]
Hi @mrkjdy I dealing with the same issue with v10.7.0
Did you manage to solve this out?

**Comment by mrkjdy** [comment]
@idanElitzur I opened a different issue #22459

**Comment by AtofStryker** [comment]
@mjhenkes do we know if the flake issues were fixed with #23297?

**Comment by mjhenkes** [comment]
These appear to have been fixed by #23297. At a minimum they cannot manifest in the same way anymore.
